Eurostatv2 pr

shapateriya opened the pull request:
Source URL: https://ec.europa.eu/eurostat/databrowser/bulk

beets commented:
Thanks for the PR and getting this import done quickly!

A few initial comments:

- Please run the formatter (./run_tests.sh -f)
- Please add details to the PR description. Specifically, I'd like to see:
   - Source file URL
   - Any commands used to generate these files. The reviewer (or anyone after) should be able to duplicate the output files based on the instructions here.
   - Any additional manual updates done as part of the import.
   - Verifications done on the output

In the future, all scripts should be accompanied by tests. We can skip it this time due to the time crunch, but I would like us to come back to revisit this in the next month.

fyi @manishvats2

beets commented on scripts/eurostat/regional_statistics_by_nuts/birth_death_migration/import_data.py:
```diff
@@ -126,8 +129,15 @@ def clean_data(preprocessed_df, output_path):
     clean_df = clean_df.replace(':', '')
-
-    clean_df['geo'] = 'dcid:nuts/' + clean_df['geo']
+    # for ind, geo in enumerate(clean_df['geo']):
```
please remove the commented out code

shapateriya replied:
removed

beets commented on scripts/eurostat/regional_statistics_by_nuts/employed_per_sector/emp_persec_preprocess_gen_tmcf.py:
```diff
@@ -37,24 +40,37 @@
     'Count_Person_Employed_NACE/O-U',
     'Count_Person_Employed_NACE/R-U',
-    'Count_Person_Employed',
+    'dc/nm9hcklgg5zb3',
```
please add a comment that this is "Population: Employed"

shapateriya replied:
added

beets commented on scripts/eurostat/regional_statistics_by_nuts/gdp/import_data.py:
```diff
@@ -72,7 +82,12 @@ def download_data(self):
         data frame.
         """
-        self.raw_df = pd.read_table(self.DATA_LINK)
+        # self.raw_df = pd.read_table(self.DATA_LINK)
```
please remove

shapateriya replied:
Removed as requested

beets commented on scripts/eurostat/regional_statistics_by_nuts/gdp/import_data.py:
```diff
@@ -72,7 +82,12 @@ def download_data(self):
+        self.raw_df = pd.read_table("nama_10r_3gdp.tsv.gz")
+        self.raw_df = self.raw_df.rename(columns=({'freq,unit,geo\TIME_PERIOD': 'unit,geo\\time'}))
+        self.raw_df['unit,geo\\time'] =  self.raw_df['unit,geo\\time'].str.slice(2)
+        # return raw_df
```
please remove

shapateriya replied:
removed

beets commented on scripts/eurostat/regional_statistics_by_nuts/gdp/import_data.py:
```diff
@@ -174,13 +191,16 @@ def generate_tmcf(self):
                     assert col in ['geo', 'time']
                     continue
                 col_num += 1
+                # Amount_EconomicActivity_GrossDomesticProduction_Nominal_AsAFractionOf_Count_Person
```
please remove

shapateriya replied:
removed .

beets commented on scripts/eurostat/regional_statistics_by_nuts/population_density/PopulationDensity_preprocess_gen_tmcf.py:
```diff
@@ -95,5 +110,9 @@ def get_template_mcf():
 
 
 if __name__ == "__main__":
+    # _DATA_URL = "https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?file=data/demo_r_d3dens.tsv.gz"
```
please remove

shapateriya replied:
removed

beets reviewed (commented):
thanks! these changes look good, especially the updates on measurement methods and stat vars. i do want us to think through the update on Count_Person_Employed --> dc/nm9hcklgg5zb3 (/cc @ajaits @hareesh-ms)

please use [git lfs](https://git-lfs.com/) for the input and output tsv / csv's. we should have examples of these in the repo

shapateriya commented:
Hi @beets ,
I've addressed your comments on [PR](https://github.com/datacommonsorg/data/pull/1094)  for Eurostat education data. Since it's outdated, I've closed it and created a new [PR ](https://github.com/datacommonsorg/data/pull/1112)reflecting the changes. Please review: [[link to PR](https://github.com/datacommonsorg/data/pull/1112)].

shapateriya commented:
I've addressed your comments on [PR](https://github.com/datacommonsorg/data/pull/1094) for the import Eurostat data. Since it's outdated, I've closed it and created a new [PR](https://github.com/datacommonsorg/data/pull/1112) reflecting the changes. Please review: [link to PR](https://github.com/datacommonsorg/data/pull/1112)

shapateriya commented:
I've addressed your comments on https://github.com/datacommonsorg/data/pull/1094 for the import Eurostat data. Since it's outdated, I've closed it and created a new https://github.com/datacommonsorg/data/pull/1112 reflecting the changes. Please review: https://github.com/datacommonsorg/data/pull/1112
